tests/python/pants_test/backend/python/tasks/native/test_ctypes_integration.py

```diff
@@ -38,45 +47,40 @@ class CTypesIntegrationTest(PantsRunIntegrationTest):
     'testprojects/src/python/python_distribution/ctypes_with_extra_compiler_flags:bin'
   )
 
-  def test_ctypes_binary_creation(self):
+  @_toolchain_variants
+  def test_ctypes_binary_creation(self, toolchain_variant):
     """Create a python_binary() with all native toolchain variants, and test the result."""
-    # TODO: this pattern could be made more ergonomic for `enum()`, along with exhaustiveness
-    # checking.
-    for variant in ToolchainVariant.allowed_values:
-      self._assert_ctypes_binary_creation(variant)
-
-  _compiler_names_for_variant = {
-    'gnu': ['gcc', 'g++'],
-    'llvm': ['clang', 'clang++'],
-  }
-
-  # All of our toolchains currently use the C++ compiler's filename as argv[0] for the linker.
-  _linker_names_for_variant = {
-    'gnu': ['g++'],
-    'llvm': ['clang++'],
-  }
-
-  def _assert_ctypes_binary_creation(self, toolchain_variant):
     with temporary_dir() as tmp_dir:
       pants_run = self.run_pants(command=['binary', self._binary_target], config={
         GLOBAL_SCOPE_CONFIG_SECTION: {
           'pants_distdir': tmp_dir,
         },
         'native-build-step': {
-          'toolchain_variant': toolchain_variant,
+          'toolchain_variant': toolchain_variant.value,
```

Comment:
I feel like we should get these be calling `str(toolchain_variant)` rather than `toolchain_variant.value`? Particularly given #7232